### PR TITLE
Fix javascript issues

### PR DESCRIPTION
**Pull request by rafalcymerys**
- Fixes issue with javascripts not loading correctly when page load finishes
- Fixes issue with incorrect type that prevented from adding to cart when multiple variants were present

### CODE REVIEW

**Comment by mrbrdo** [comment]
@rafalcymerys see also this commit:
https://github.com/mrbrdo/spree_legacy_frontend/commit/5273c3f184ef6c217a6a8f753edfdfc4a5d530dc

Different versions of spree.js are problematic if using 4.4+ version of Spree (core). (not sure if I did any change in spree_backend for this too)

**Comment by rafalcymerys** [comment]
@mrbrdo thank you for keeping an eye out :) I merged this fix for conflicts between spree_frontend and spree_backend yesterday, let me know if that's the same issue you mentioned: https://github.com/spree/spree_legacy_frontend/pull/21

**Comment by mrbrdo** [comment]
@rafalcymerys yes it is, and also fixed more properly with the `spree/frontend/api/main`. Nice to see progress on the PRs, I will check if I have any more stuff to make into PRs so we don't do stuff twice :)
In the meantime you can keep in mind my branch https://github.com/mrbrdo/spree_legacy_frontend/tree/4_4_fixes
So as to not repeat the same work. I am using it in production with Spree core main branch (4.5.x).

File: app/assets/javascripts/spree/frontend/main.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 function Spree () {}
 
 Spree.ready = function (callback) {
-  return jQuery(document).on('page:load turbo:load', function () {
+  return jQuery(document).on('DOMContentLoaded page:load turbo:load', function () {
```
Comment (mrbrdo) [inline app/assets/javascripts/spree/frontend/main.js:5]: Not sure this change is needed anymore after https://github.com/spree/spree_legacy_frontend/pull/21

**Comment by rafalcymerys** [comment]
I guess we won't need this one anymore, since it was resolved by changing the project setup process.

**Pull request closed without merging**